v0/destinations/appcues/transform.js

```diff
@@ -0,0 +1,184 @@
+  Object.keys(mappingJson).forEach(destKey => {
+    let value = get(message, mappingJson[destKey]);
+    if (value) {
+      eventJson[destKey] = value;
+    }
+  });
+  return eventJson;
+}
+
+function processTrackEvent(message, destination, mappingJson) {
+  let requestJson = {};
+  if (message.messageId) {
+    requestJson.request_id = message.messageId;
+  }
+
+  let eventJson = getTrackPayload(message, mappingJson);
+  eventJson.timestamp = utcTimeToEpoch(eventJson.timestamp);
+  requestJson.events = [eventJson];
+
+  return buildResponse(
+    message,
+    requestJson,
+    getEndpoint(destination.Config.accountId, message.userId)
+  );
+
+}
+
+function getProfilePayload(message) {
+  let profileJson = {};
+  const category = ConfigCategory.PROFILE;
+  const mappingJson = mappingConfig[category.name];
+  Object.keys(mappingJson).forEach(destKey => {
+    if (typeof mappingJson[destKey] === "object") {
+      const { sourceKeys } = mappingJson[destKey];
+      const value = getValueFromMessage(message, sourceKeys);
+      if(!!value)
+      {
+      set(profileJson, destKey, value);
+      }
+    } else {
+      set(profileJson, destKey, get(message, mappingJson[destKey]));
+    }
+  });
+  return profileJson;
+}
+
+function processPage(message, destination, mappingJson) {
+  
+  let requestJson = {};
+  if (message.messageId) {
+    requestJson.request_id = message.messageId;
+  }
+
+  // generating profile part of the payload
+  let profileJson = getProfilePayload(message);
+  profileJson._appcuesId = destination.Config.accountId;
+  requestJson.profile_update = profileJson;
+  
+  // generating event part of the payload
+  let eventJson = getTrackPayload(message, mappingJson);
+  eventJson.timestamp = utcTimeToEpoch(eventJson.timestamp);
+  eventJson.name = "Viewed a Page";
+  eventJson.attributes._identity = {};
+  eventJson.attributes._identity.userId = message.userId;
+  eventJson.attributes = removeUndefinedAndNullAndEmptyValues(eventJson.attributes);
+  if((message.properties && message.properties.url) || (message.context.page && message.context.page.url))
+  {
+  eventJson.context = {};
+  const sourceKeys = ["context.page.url", "properties.url"];
+  eventJson.context.url = getValueFromMessage(message, sourceKeys);
+  }
+  requestJson.events = [eventJson];
+
+  return buildResponse(
+    message, 
+    requestJson, 
+    getEndpoint(destination.Config.accountId, message.userId)
+    );
+}
+
+function process(event) {
+  const respList = [];
+  let response;
+  const {
+    message,
+    destination
+  } = event;
+
+  if (!message.type) {
+    throw new Error("Message Type is not present. Aborting message.");
+  }
+
+  const messageType = message.type.toLowerCase();
+
+  let category = ConfigCategory.TRACK;
+
+  switch (messageType) {
+    case EventType.TRACK:
+      response = processTrackEvent(
+        message,
+        destination,
+        mappingConfig[category.name]
+      );
+      break;
+    case EventType.PAGE:
```

Comment:
Through cloud-mode as we are suppporting type `page`, we should support `screen` as well in the same way

Reply:
Added it as a track call with name `Viewed a Screen` along with properties of screen call